recipes/mongo-cxx-driver/all/conanfile.py

```diff
@@ -0,0 +1,117 @@
+        # dependencies
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-r" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["BSONCXX_POLY_USE_MNMLSTC"] = self.options.polyfill == "mnmlstc"
+        self._cmake.definitions["BSONCXX_POLY_USE_STD_EXPERIMENTAL"] = self.options.polyfill == "experimental"
+        self._cmake.definitions["BSONCXX_POLY_USE_BOOST"] = self.options.polyfill == "boost"
+        self._cmake.definitions["BUILD_VERSION"] = self.version
+        self._cmake.definitions["BSONCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        self._cmake.definitions["MONGOCXX_LINK_WITH_STATIC_MONGOC"] = not self.options["mongo-c-driver"].shared
+        # FIXME: two CMake module/config files should be generated (mongoc-1.0-config.cmake and bson-1.0-config.cmake),
+        # but it can't be modeled right now.
+        # Fix should happen in mongo-c-driver recipe
+        shutil.copy("Findmongoc-1.0.cmake", "Findbson-1.0.cmake")
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def build(self):
+        if self.settings.compiler == "Visual Studio":
+            cmake_file = os.path.join(self._source_subfolder, "../CMakeLists.txt")
+            tools.replace_in_file(
+                cmake_file,
+                'add_subdirectory("source_subfolder")',
+                'add_subdirectory("source_subfolder")\nadd_definitions(-D_ENABLE_EXTENDED_ALIGNED_STORAGE)'
+            )
+
+        self._patch_sources()
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD-PARTY-NOTICES", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        # Need to ensure mongocxx is linked before bsoncxx
+        self.cpp_info.libs = sorted(tools.collect_libs(self), reverse=True)
+        self.cpp_info.includedirs.extend([os.path.join("include", lib, "v_noabi") for lib in ["bsoncxx", "mongocxx"]])
+
+        if self.options.polyfill == "mnmlstc":
+            self.cpp_info.includedirs.append(os.path.join("include", "bsoncxx", "third_party", "mnmlstc"))
+
+        if not self.options.shared:
+            self.cpp_info.defines.extend(["BSONCXX_STATIC", "MONGOCXX_STATIC"])
```

Comment:
Probably not required, since upstream uses generate_export_header:
- https://github.com/mongodb/mongo-cxx-driver/blob/5b4628114ebddc9bd0092ea0eb3d66beef7818e9/cmake/BsoncxxUtil.cmake#L47
- https://github.com/mongodb/mongo-cxx-driver/blob/5704837c094f322df7bf4ef32b0809655ae47b95/cmake/MongocxxUtil.cmake#L42

Reply:
Actually, it is required.

I added these lines to test package:
```
#ifndef BSONCXX_STATIC
static_assert(false);
#endif
```

And compiled the static library. It compiled without any problem.

But when I removed the lines you suggested, the compilation failed.